cms/djangoapps/api/docs/decisions/001-course-apps-api.rst

```diff
@@ -0,0 +1,153 @@
+=======
+
+The new `Course Authoring MFE`_ includes a new UX called "Pages and Resources"
+for configuring course "apps" such as progress, wiki, teams, discussions etc.
+all from one place. Currently these apps are intermingled with the core
+platform and are inseparable from it, however future apps could be implemented
+as plugins.
+
+In order to drive this new UI, we need a way to discover installed and available
+course apps and provide an API that can be used by MFEs to display information
+related to these apps and potentially enable/disable these apps.
+
+.. _Course Authoring MFE: https://github.com/edx/frontend-app-course-authoring/
+
+
+Decision
+========
+
+We can develop a new course apps API that can be used to manage the apps
+available for a course. The list of *available* apps might not match the list
+of *installed* apps since it is possible that certain apps may be disabled, or
+otherwise unavailable for a particular course.
+
+The course apps API will only list the available apps, and allow toggling them
+on or off. It will not support any other verbs or settings. Each app can
+provide its own API(s) for configuring app-specific settings, which will not be
+in scope here. Currently the mechanism for extending the frontend with custom
+config pages for plug-in apps is TBD.
+
+Each app will have the following associated metadata:
+
+- **id**: A unique identifier for the app. This can be used in the frontend to
+  look up the app's friendly title and description.
+- **enabled**: Is this app enabled for the current course.
+- **permissions**: Apps can potentially enable/disable certain operations, for
+  users based on their role. The permissions a user has can be listed here. Apps
+  may have their own permissions but at this very least this will include:
+
+    - **enable**: Can the current user enable/disable this app.
+    - **configure**: Can the current user can configure this app.
+
+  Apps may have their own custom fine-grained controls over what settings a
+  particular user can configure.
+
+  If an app doesn't have any configuration it can leave out the ``configure``
+  key and the UI will simply not show a configuration option for that app.
+- **legacy_link**: If available, this will point to the legacy link for
+  configuring the app. This can be provided as a fallback while the new UX is
+  still in development.
+
+Here is the structure that will form the basis of the API's response:
+
+.. code-block:: python
+
+    {
+        'id': 'courseapp',
+        'enabled': False,
+        'permissions': {
+            'enable': True,
+            'configure': True,
+            'edit_lti_config': False,
+        },
+        'legacy_link': 'https://studio.example.com/course_id/app-page',
+    }
+
+
+This API can be hosted at: ``/course_apps/v1/apps/{course_id}/``
+
+A ``GET`` request to this API will return an array of objects with the above
+structure. A ``PATCH`` request to the same endpoint with just the ``id`` and the
```

Comment:
Do you think it makes sense for `/course_apps/v1/apps/{course_id}/{app_id}` to manage the enabled/disabled state?  Or should app-specific endpoints (outside of this set of APIs) manage that state?  Are we essentially saying that enabled/disabled is a feature/config option for the course apps system, not for the individual apps themselves?  i.e., enabled/disabled state for all apps is stored in one table together, while the app-specific config is probably stored elsewhere?

I don't feel strongly, more curious if that was the intention here.

Reply:
The intention was for the enabled/disabled state to be managed by the course apps system. Although an app could be disabled due to being misconfigured etc. 

The app configuration could be managed by the apps themselves through each apps individual API.